### PR TITLE
chore: Python codebase refactor

### PR DESCRIPTION
**Pull request by koconder**
## Description

Light refactor of python codebase with best practice, strings are escaped and some minor code tweaks.

## How was this patch tested?

Locally tested using existing tests and codebase.

## Does this PR introduce _any_ user-facing changes?

No

### CODE REVIEW

**Comment by koconder** [comment]
I have patched the broken flake8 PEP 8 codestyle errors in unit testing

**Comment by koconder** [comment]
@allisonport-db @scottsand-db @tdas @vkorukanti mind reviewing this light refactoring PR?

**Comment by scottsand-db** [comment]
Hi @koconder - thanks for the ping. Yup I will be taking a look at it, just a little busy right now working on the next release of Delta. Will get to this soon! Thanks!

File: examples/python/quickstart_sql_on_paths.py [hunk 1]
```diff
@@ -41,22 +41,23 @@
...
 
-    spark.sql("SELECT * FROM delta.`%s`" % table_dir).show()
+    spark.sql(f"SELECT * FROM delta.`{table_dir}`").show()
 
     # Update table data
     print("########## Overwrite the table ###########")
-    spark.sql("INSERT OVERWRITE delta.`%s` select * FROM (VALUES 5, 6, 7, 8, 9) x (id)" % table_dir)
-    spark.sql("SELECT * FROM delta.`%s`" % table_dir).show()
+    spark.sql(f"INSERT OVERWRITE delta.`{table_dir}` select * FROM (VALUES 5, 6, 7, 8, 9) x (id)")
+
```
Comment (scottsand-db) [inline examples/python/quickstart_sql_on_paths.py:49]: nit: delete empty line

File: examples/python/utilities.py [hunk 2]
```diff
@@ -59,7 +59,7 @@
 
 # SQL describe history
 print("####### SQL Describe History ########")
-print(spark.sql("DESCRIBE HISTORY delta.`%s`" % ("/tmp/delta-table")).collect())
+print(spark.sql('DESCRIBE HISTORY delta.`/tmp/delta-table`').collect())
```
Comment (scottsand-db) [inline examples/python/utilities.py:62]: nit: why the change from `"` to `'` ? you haven't done it so far in this PR?

File: python/delta/tables.py [hunk 3]
```diff
@@ -624,27 +624,29 @@ def _dict_to_jmap(
...
 
         jvm: "JVMView" = sparkSession._sc._jvm  # type: ignore[attr-defined]
 
         jmap: "JavaMap" = jvm.java.util.HashMap()
         for col, expr in pydict.items():
             if type(col) is not str:
-                e = ("Keys of dict in %s must contain only strings with column names" % argname) + \
-                    (", found '%s' of type '%s" % (str(col), str(type(col))))
+                e = f"Keys of dict in {argname} must contain only strings with column names" + \
+                    ", found '%s' of type '%s" % (str(col), str(type(col)))
```
Comment (scottsand-db) [inline python/delta/tables.py:638]: why not use `f"..."` and `{str(col)}` here, too?

File: python/delta/tables.py [hunk 4]
```diff
@@ -624,27 +624,29 @@ def _dict_to_jmap(
...
             if type(expr) is Column:
                 jmap.put(col, expr._jc)
             elif type(expr) is str:
                 jmap.put(col, functions.expr(expr)._jc)
             else:
-                e = ("Values of dict in %s must contain only Spark SQL Columns " % argname) + \
+                e = f"Values of dict in {argname} must contain only Spark SQL Columns" + \
                     "or strings (expressions in SQL syntax) as values, " + \
-                    ("found '%s' of type '%s'" % (str(expr), str(type(expr))))
+                    "found '%s' of type '%s'" % (str(expr), str(type(expr)))
```
Comment (scottsand-db) [inline python/delta/tables.py:648]: ditto

**Review by scottsand-db — CHANGES_REQUESTED** [review]
LGTM with minor nits. @vkorukanti take a look?

**Comment by koconder** [comment]
@scottsand-db comments and issues resolved

**Comment by scottsand-db** [comment]
@koconder - after discussing with another engineer @vkorukanti on my team, this PR is a little bit too big to review. Could you split this up into, say, 2 or 3 smaller PRs?

**Comment by vkorukanti** [comment]
> @koconder - after discussing with another engineer @vkorukanti on my team, this PR is a little bit too big to review. Could you split this up into, say, 2 or 3 smaller PRs?

May be one file per PR? Sorry, it is more work, but helps with review and testing.

**Comment by scottsand-db** [comment]
As mentioned above, we've asked that you kindly split up this PR into smaller PRs, as that will be easier to review. Closing this PR for now. Cheers.

**Pull request closed without merging**